2.17 Migration error for item_category_join

## What went wrong? 😲

@raviSussol called this https://github.com/msupply-foundation/open-msupply/pull/10486#discussion_r2840137231, and I just ran into it:

```
Running `target/debug/remote_server`
2026-02-25 14:13:46.253023000 [INFO ] <server:102>:Development server starting on port 8000
2026-02-25 14:13:46.253385000 [INFO ] <service::sync:175>:Running as central from override
2026-02-25 14:13:46.253598000 [INFO ] <repository::database_settings:195>:Connecting to database 'omsupply-database.sqlite'
2026-02-25 14:13:46.266266000 [INFO ] <server:116>:Run DB migrations...
2026-02-25 14:13:46.271740000 [INFO ] <repository::migrations::views:75>:Dropping database views...
2026-02-25 14:13:46.275927000 [INFO ] <repository::migrations:228>:Running one time database migration 2.16.0
2026-02-25 14:13:46.277854000 [INFO ] <repository::migrations:250>:Running database migration fragment version 2.16.0: log_tag_sensor_type
2026-02-25 14:13:46.278575000 [INFO ] <repository::migrations:250>:Running database migration fragment version 2.16.0: add_migration_and_server_status_to_system_log_type_enums
2026-02-25 14:13:46.278634000 [INFO ] <repository::migrations:250>:Running database migration fragment version 2.16.0: add_shipping_method_id_to_invoice
2026-02-25 14:13:46.281386000 [INFO ] <repository::migrations:250>:Running database migration fragment version 2.16.0: add_version_to_plugins
2026-02-25 14:13:46.285508000 [INFO ] <repository::migrations:250>:Running database migration fragment version 2.16.0: available_volume_on_requisition_line_and_volume_per_pack_on_item
2026-02-25 14:13:46.294499000 [INFO ] <repository::migrations:250>:Running database migration fragment version 2.16.0: add_universal_code_to_item
2026-02-25 14:13:46.296469000 [INFO ] <repository::migrations:250>:Running database migration fragment version 2.16.0: add_comments_required_to_asset_log_reason
2026-02-25 14:13:46.298336000 [INFO ] <repository::migrations:228>:Running one time database migration 2.17.0
2026-02-25 14:13:46.298411000 [INFO ] <repository::migrations:250>:Running database migration fragment version 2.17.0: item_category_join_add_item_link_id
2026-02-25 14:13:46.298791000 [ERROR] <server:121>:Failed to run DB migrations: Error during fragment time migration (2.17.0) (item_category_join_add_item_link_id) -> [
    "Sql error \n                PRAGMA foreign_keys = OFF;\n                ALTER TABLE item_category_join\n                ADD COLUMN item_link_id TEXT NOT NULL REFERENCES item_link (id) DEFAULT 'temp_for_migration'; \n                UPDATE item_category_join SET item_link_id = item_id;\n                PRAGMA foreign_keys = ON;\n         ",
    "UNKNOWN (\"Cannot add a REFERENCES column with non-NULL default value      0\")",
]
```
## Expected behaviour 🤔

Safe and successful migration :)

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. have a 2.16 sqlite db, I think requiring some existing item_category_join records.
2. Upgrade to 2.17 dev e54583d999

## Your environment 🌱

Mac dev sqlite

--- a/server/repository/src/migrations/v2_17_00/item_category_join_add_item_link_id.rs
+++ b/server/repository/src/migrations/v2_17_00/item_category_join_add_item_link_id.rs
@@ -26,14 +26,25 @@ impl MigrationFragment for Migrate {
         sql!(
             connection,
             r#"
-                PRAGMA foreign_keys = OFF;
-                ALTER TABLE item_category_join
-                ADD COLUMN item_link_id TEXT NOT NULL REFERENCES item_link (id) DEFAULT 'temp_for_migration'; 
-                UPDATE item_category_join SET item_link_id = item_id;
-                PRAGMA foreign_keys = ON;
+                CREATE TABLE item_category_join_new (
+                    id TEXT PRIMARY KEY NOT NULL,
+                    item_link_id TEXT NOT NULL REFERENCES item_link(id),
+                    category_id TEXT NOT NULL REFERENCES category(id),
+                    deleted_datetime TEXT
+                );
+
+                INSERT INTO item_category_join_new (id, item_link_id, category_id, deleted_datetime)
+                SELECT id, item_id, category_id, deleted_datetime
+                FROM item_category_join;
+
+                DROP TABLE item_category_join;
+                ALTER TABLE item_category_join_new RENAME TO item_category_join;
+
+                CREATE INDEX "index_item_category_join_item_link_id_fkey" ON "item_category_join" ("item_link_id");
          "#,
         )?;
 
+        #[cfg(feature = "postgres")]
         sql!(
             connection,
             r#"

